chapter_preliminaries/linear-algebra.md

```diff
@@ -966,42 +1027,62 @@ In deep learning, we are often trying to solve optimization problems:
+of photos of different people. 
+These distances, which constitute 
+the objectives of deep learning algorithms, 
+are often expressed as norms. 
 
 
 ## Summary
 
-In just this section,
-we have taught you all the linear algebra
+In this section, we reviewed all the linear algebra
 that you will need to understand
 a remarkable chunk of modern deep learning.
 There is a lot more to linear algebra
-and a lot of that mathematics is useful for machine learning.
+and much of it is useful for machine learning.
 For example, matrices can be decomposed into factors,
 and these decompositions can reveal
 low-dimensional structure in real-world datasets.
 There are entire subfields of machine learning
 that focus on using matrix decompositions
 and their generalizations to high-order tensors
-to discover structure in datasets and solve prediction problems.
+to discover structure in datasets 
+and solve prediction problems.
 But this book focuses on deep learning.
-And we believe you will be much more inclined to learn more mathematics
+And we believe you will be more inclined 
+to learn more mathematics
 once you have gotten your hands dirty
-deploying useful machine learning models on real datasets.
-So while we reserve the right to introduce more mathematics much later on,
-we will wrap up this section here.
-
-If you are eager to learn more about linear algebra,
-you may refer to either a number of excellent books and online resources :cite:`Strang.1993,Kolter.2008,Petersen.Pedersen.ea.2008` or read more in our
-[appendix on linear algebra](https://d2l.ai/chapter_appendix-mathematics-for-deep-learning/geometry-linear-algebraic-ops.html).
-
-For now, the following will suffice:
-* Scalars, vectors, matrices, and tensors of are the basic mathematical objects used in linear algebra. In particular, vectors generalize scalars, matrices generalize vectors, and lastly, tensors generalize matrices. 
-* Scalars, vectors, matrices, and tensors have zero, one, two, and an arbitrary number of axes, respectively.
-* A tensor can be sliced or it can be reduced along the specified axes by operations such as `sum` and `mean`.
-* Elementwise multiplication of two matrices is called their Hadamard product. It is different from matrix multiplication both in terms of effect (elementwise) and speed (quadratic rather than cubic time to compute).
-* In deep learning, we often work with norms such as the $L_1$ norm, the $L_2$ norm, and the Frobenius norm.
+applying machine learning to real datasets.
+So while we reserve the right 
+to introduce more mathematics later on,
+we wrap up this section here.
+
+If you are eager to learn more linear algebra,
+there are many excellent books and online resources.
+For a more advanced crash course, consider checking out
+:cite:`Strang.1993,Kolter.2008,Petersen.Pedersen.ea.2008` 
+or our [online appendix on linear algebra](https://d2l.ai/chapter_appendix-mathematics-for-deep-learning/geometry-linear-algebraic-ops.html).
+
+To recap:
```

Comment:
add space this line:

now it's not rendering properly http://preview.d2l.ai/d2l-en/PR-1856/chapter_preliminaries/linear-algebra.html#dot-products